yaksh/models.py

```diff
@@ -1077,11 +1083,16 @@ def regrade(self, question_id):
         json_data = question.consolidate_answer_data(answer) \
                             if question.type == 'code' else None
         correct, result = self.validate_answer(answer, question, json_data)
-        user_answer.marks = question.points if correct else 0.0
         user_answer.correct = correct
         user_answer.error = result.get('error')
+        if correct:
```

Comment:
I agree with @prathamesh920 that each test case having marks is non optimal.  I think I have a nicer way to specify this.  Basically tread the "marks" you have above as a weight for each test, then the total mark is sum(obtained weight)/total(weight)*marks.  By default each test case has a weight of 1, this way is nicer I think and much easier to implement.  A teacher may even change the total marks without having to worry about the individual weights.